### ISSUE TITLE
Add grid for metadata report results

### ISSUE THREAD

**Issue by rctauber** [OBOFoundry/OBOFoundry.github.io#779]
Each ontology should have a result from the metadata validation: pass, warning, or fail. Currently, the system only has pass or fail, with a few ontologies not included in the checks. The status of the metadata should be visible on the front page of the OBO registry website.

**Proposal:**
An ontology will receive a `pass` when it passes the full `metadata-schema.json` check.
An ontology will receive a `warning` when it fails the full check, but passes the `metadata-schema-lite.json` check.
An ontology will receive a `fail` when it fails both checks.

`metadata-schema.json` includes restrictions on licenses. In the future, we can make the `github` contact information required as well.

**Comment by bpeters42** [comment]
We will implement a new status 'failing validation', working just like obsolete, meaning an incomplete registry entry is allowed for these. Like the obsoletes, these will be displayed at the bottom of the list. For non-failing, non-obsolete, non-orphan ontologies, all pull requests will have to the metadata report with 'pass' or 'warning'. For the warnings, we will develop a system that classifies them into critical (non-standard license) vs. 'info' (like no publication associated), and decide on how to display them in the future.

**Comment by rctauber** [comment]
Work on branch: https://github.com/rctauber/OBOFoundry.github.io/tree/metadata-grid

The validation process outputs two products. The first is the violation report, which only contains only the error messages (these are also logged in the console so you can easily see an issue in the Travis build). The second is a metadata grid with all fields, for example:

![image](https://user-images.githubusercontent.com/16750150/53896548-109a7a80-3fe9-11e9-8466-7c90aecc38e4.png)

These results are sorted in the following order:
1. Foundry status (true, false)
2. Ontology activity status (active, orphaned, inactive)
3. Validation status (pass, info, warn, fail)
4. Alphabetical

**Comment by cmungall** [comment]
@dougli1sqrd @kltm this is analogous to the checks we do on GO annotation sources, isn't it (of course, the OBO check is ontologies rather than annotations)? We have checks on one axis and source on the other. Is there any code that can be reused?

@rctauber do we plan to meld the robot report and reasoning checks into the same summary table? If so I expect we'll want to do this in the OBO jenkins job.

**Comment by jamesaoverton** [comment]
Note that in this grid, most of the columns correspond directly to a JSON schema file: https://github.com/OBOFoundry/OBOFoundry.github.io/tree/master/util/schema

TLDR: These JSON schemas are primarily for Travis checks on this repository. But yes, we should have a Jenkins job that can merge these checks with ROBOT reports somehow.

We currently have:

- these OBO registry metadata checks (JSON schema, run by Python scripts against YAML/Markdown files in the OBO registry, especially important for our Travis checks that help maintain the quality of the registry)
- ROBOT reports (SPARQL queries run against OWL files, especially important to check the quality of ontology releases)

There are important differences between the metadata for

1. ontology projects
2. ontology releases
3. ontology terms

Last year I made this Google Sheet based on an email from Bjoern, trying to pull them apart: https://docs.google.com/spreadsheets/d/18i_3fzfhl8-lWKahU7tswgJIf6RLyosKRY1DMXbDHH0/edit#gid=0

Eventually we need to bring together ROBOT reports on the ontology project metadata included in each ontology release (i.e. published OWL file) to ensure that they match up.

**Comment by rctauber** [comment]
The next step is to update the homepage using this grid to sort the entries (see #830) and create an HTML version of this grid with colors.

**Comment by jamesaoverton** [comment]
Implemented in #842.
